src/librustc_resolve/diagnostics.rs

```diff
@@ -904,6 +904,10 @@ impl<'a> Resolver<'a> {
         );
         self.add_typo_suggestion(err, suggestion, ident.span);
 
+        let import_suggestions =
+            self.lookup_import_candidates(ident, Namespace::MacroNS, parent_scope, |_| true);
```

Comment:
You could modify the `|_| true` to check its argument, which is a `Res`, with something like the following to only suggest macros that _can_ be called like `foo!()`, which should solve the "suggest already present macro" problem:

```
match res {
    Res::Def(DefKind::Macro(MacroKind::Bang), _) => true,
    _ => false,
}
```
}

Reply:
This removes the bad error message, but means there's no suggestions if it's a proc macro or attribute. It would be nice to say something like 'expected `macro_rules` macro, but got a proc macro'. Maybe that belongs in a follow-up PR though.

Reply:
We can handle this in a follow up PR.

Reply:
This is https://github.com/rust-lang/rust/issues/76429 I think.